pkg/apis/core/types.go

```diff
@@ -2418,16 +2418,19 @@ type ResourceResizeRestartPolicy string
 
 // These are the valid resource resize restart policy values:
 const (
-	// 'NotRequired' means Kubernetes will try to resize the container
-	// without restarting it, if possible. Kubernetes may however choose to
-	// restart the container if it is unable to actuate resize without a
-	// restart. For e.g. the runtime doesn't support restart-free resizing.
-	NotRequired ResourceResizeRestartPolicy = "NotRequired"
-	// 'RestartContainer' means Kubernetes will resize the container in-place
+	// ResizeRestartPolicyPreferNoRestart means Kubernetes will try to resize the container without
+	// restarting it, if possible. This option does not guarantee that the container won't be
+	// restarted during the resize.
+	// This is the implicit default when no resource resize policy is specified.
+	ResizeRestartPolicyPreferNoRestart ResourceResizeRestartPolicy = "PreferNoRestart"
```

Comment:
Introducing a new value for a field is ~the same introducing a new field.  It has to begin in alpha.  Presuming this KEP goes beta in 33, there's a very small chance that someone upgrades to 33 (where the feature is on by default), uses this new value, and then downgrades to 32, where the saved object will fail validation.

So I think you need a new gate for this.